auth-web/src/components/auth/manage-business/AddBusinessDialog.vue

```diff
@@ -172,238 +174,292 @@
+
+    const passcodeMaxLength = computed(() => {
+      if (isFirm.value) return 150
+      if (isCooperative.value) return 9
+      return 15
+    })
+
+    const passcodeRules = computed(() => {
+      if (isFirm.value) {
+        return [
+          (v) => !!v || 'Proprietor or Partner Name is required',
+          (v) => v.length <= 150 || 'Maximum 150 characters'
+        ]
+      }
+      if (isCooperative.value) {
+        return [
+          (v) => !!v || 'Passcode is required',
+          (v) => CommonUtils.validateCooperativePasscode(v) || 'Passcode must be exactly 9 digits'
+        ]
+      }
       return [
-        v => !!v || 'Passcode is required',
-        v => CommonUtils.validateCooperativePasscode(v) || 'Passcode must be exactly 9 digits'
+        (v) => !!v || 'Password is required',
+        (v) => CommonUtils.validateCorporatePassword(v) || 'Password must be 8 to 15 characters'
       ]
-    }
-    return [
-      v => !!v || 'Password is required',
-      v => CommonUtils.validateCorporatePassword(v) || 'Password must be 8 to 15 characters'
-    ]
-  }
-
-  get forgotButtonText (): string {
-    return 'I lost or forgot my ' + (this.isCooperative ? 'passcode' : 'password')
-  }
+    })
+
+    const forgotButtonText = computed(() => {
+      return 'I lost or forgot my ' + (isCooperative.value ? 'passcode' : 'password')
+    })
+
+    const helpDialogBlurb = computed(() => {
+      if (isCooperative.value) {
+        return 'If you have not received your Access Letter from BC Registries, or have lost your Passcode, ' +
+          'please contact us at:'
+      } else {
+        const url = 'www.corporateonline.gov.bc.ca'
+        return `If you have forgotten or lost your password, please visit <a href="https://${url}">${url}</a> ` +
+          'and choose the option "Forgot Company Password", or contact us at:'
+      }
+    })
 
-  get helpDialogBlurb (): string {
-    if (this.isCooperative) {
-      return 'If you have not received your Access Letter from BC Registries, or have lost your Passcode, ' +
-        'please contact us at:'
-    } else {
-      const url = 'www.corporateonline.gov.bc.ca'
-      return `If you have forgotten or lost your password, please visit <a href="https://${url}">${url}</a> ` +
-        'and choose the option "Forgot Company Password", or contact us at:'
+    const isFormValid = computed(() => {
+      // if user is a staff user or sbc staff user, then only require the business identifier
+      if (props.isStaffOrSbcStaff && !!businessIdentifier.value) {
+        return true
+      }
+      // business id is required
+      // passcode is required
+      // firms must accept certify clause
+      // staff users must enter names
+      // validate the form itself (according to the components' rules/state)
+      return (
+        !!businessIdentifier.value &&
+        !!passcode.value &&
+        (!isFirm.value || isCertified.value) &&
+        (!!certifiedBy.value || !isFirm.value) &&
```

Comment:
(!isFirm.value || isCertified.value) && (!!certifiedBy.value || !isFirm.value) seem to have some redundancy. If isFirm.value is false, both conditions will be true, regardless of the other values.

Reply:
Should have been addressed in https://github.com/bcgov/sbc-auth/pull/2403

Reply:
`        (!isFirm.value || (isCertified.value && !!certifiedBy.value)) &&`